index.js

```diff
@@ -50,6 +50,11 @@ module.exports = exports = function fetch(input, init = {}) {
       request._headers.set('user-agent', `Bare/${Bare.version.substring(1)}`)
     }
 
+    if (crossOriginRedirect) {
+      request.headers.delete('authorization')
+      request.headers.delete('www-authenticate')
+    }
```

Comment:
We should store the headers in a list and add a reference to https://fetch.spec.whatwg.org/#cors-non-wildcard-request-header-name. Why is `www-authenticate` in there by the way? It's not defined as a CORS non-wildcard request-header name.

Reply:
Saw it here: https://github.com/node-fetch/node-fetch/blob/8b3320d2a7c07bce4afc6b2bf6c3bbddda85b01f/src/index.js#L210, nothing more. I've got more familiar to this spec now, I didn't create the "unsafe headers list" to keep the implementation minimal.